tests/test_routes.py

```diff
@@ -253,3 +253,210 @@ def test_query_order_list_by_category(self):
+        """Create a new Order"""
+        test_order = OrderFactory()
+        logging.debug(test_order)
+        resp = self.app.post(
+            BASE_URL, json=test_order.serialize(), content_type=CONTENT_TYPE_JSON
+        )
+        self.assertEqual(resp.status_code, status.HTTP_201_CREATED)
+        # Make sure location header is set
+        location = resp.headers.get("Location", None)
+        self.assertIsNotNone(location)
+        # Check the data is correct
+        new_order = resp.get_json()
+        self.assertEqual(new_order["name"], test_order.name, "Names do not match")
+        self.assertEqual(
+            new_order["category"], test_order.category, "Categories do not match"
+        )
+        self.assertEqual(
+            new_order["available"], test_order.available, "Availability does not match"
+        )
+        # Check that the location header was correct
+        resp = self.app.get(location, content_type=CONTENT_TYPE_JSON)
+        self.assertEqual(resp.status_code, status.HTTP_200_OK)
+        new_order = resp.get_json()
+        self.assertEqual(new_order["name"], test_order.name, "Names do not match")
+        self.assertEqual(
+            new_order["category"], test_order.category, "Categories do not match"
+        )
+        self.assertEqual(
+            new_order["available"], test_order.available, "Availability does not match"
+        )
+
+    # def test_create_order_no_data(self):
+    #     """Create a Order with missing data"""
+    #     resp = self.app.post(BASE_URL, json={}, content_type=CONTENT_TYPE_JSON)
+    #     self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    # def test_create_order_no_content_type(self):
+    #     """Create a Order with no content type"""
+    #     resp = self.app.post(BASE_URL)
+    #     self.assertEqual(resp.status_code, status.HTTP_415_UNSUPPORTED_MEDIA_TYPE)
+
+    # def test_create_order_bad_available(self):
+    #     """ Create a Order with bad available data """
+    #     test_order = OrderFactory()
+    #     logging.debug(test_order)
+    #     # change available to a string
+    #     test_order.available = "true"
+    #     resp = self.app.post(
+    #         BASE_URL, json=test_order.serialize(), content_type="application/json"
+    #     )
+    #     self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    # def test_create_order_bad_gender(self):
+    #     """ Create a Order with bad available data """
+    #     order = OrderFactory()
+    #     logging.debug(order)
+    #     # change gender to a bad string
+    #     test_order = order.serialize()
+    #     test_order["gender"] = "male"    # wrong case
+    #     resp = self.app.post(
+    #         BASE_URL, json=test_order, content_type="application/json"
+    #     )
+    #     self.assertEqual(resp.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_update_order(self):
+        """Update an existing Order"""
+        # create a order to update
```

Comment:
This section is for another issue
- Update Order
- Delete Order

Reply:
I deleted this part and write new code based on the sample code.